client/src/api/v2/camera.js

```diff
@@ -204,19 +211,9 @@ export function useVideoFeedStatus() {
  * @param {string} device Device
  * @returns {CameraStatus} Camera status
  */
-export function useCameraStatus(device) {
-  // Only run init once per render
-  useEffect(() => {
-    emit(`status-camera-${device}`);
-  }, [device]);
-
-  const [state, setState] = useState(false);
-
-  const handler = useCallback((newPayload) => {
-    setState(newPayload.connected);
-  }, []);
-
-  useChannel(`status-camera-${device}`, handler);
-
-  return state;
-}
+export const useCameraStatus = (device) => createStatusPayloadHook(
+  device, {
+    initValue: false,
+    payloadHandler: (setState) => (newPayload) => setState(newPayload?.connected ?? false)
```

Comment:
Its the very latest javascript operators. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
Side note. @haotongwang definitely does functional programming LOL

Reply:
https://medium.com/javascript-scene/curry-and-function-composition-2c208d774983

Reply:
I definitely feel like curry functions are an advanced technique (you're just too good @haotongwang). The only place that I HAVE seen curried functions is when you work with selectors via react + redux. In other normal usage, I haven't seen it used, purely because you will get the same reaction as @hallgchris did. Really up to the team to decide if this is something we want to use within the codebase.

Reply:
@khanguslee All from FIT2102 haha. I can change it to not use curried functions but it just makes the code less clean but more understandable for people not familiar with curried functions. I'll see if I can remove the curried functions without making the implementation too messy.

Reply:
Curious as to what the implementation looks like without curried functions @haotongwang

Reply:
I think the new operators are useful and makes it much more understandable (e.g. `p ?? initValue` would have to be replaced with `p === null && initvalue`.

I've removed the curried function for `payloadHandler`. The code's a bit messier but I agree curried functions might be a bit too much...
6e427c4